src/modules/kvs/kvs.c

```diff
@@ -714,13 +714,23 @@ static void content_store_completion (flux_future_t *f, void *arg)
 {
     kvs_ctx_t *ctx = arg;
     struct cache_entry *entry;
-    const char *blobref;
+    const char *cache_blobref, *blobref;
     int ret;
 
+    cache_blobref = flux_future_aux_get (f, "cache_blobref");
+    assert (cache_blobref);
+
     if (flux_content_store_get (f, &blobref) < 0) {
         flux_log_error (ctx->h, "%s: flux_content_store_get", __FUNCTION__);
         goto done;
     }
+
+    if (strcmp (blobref, cache_blobref)) {
```

Comment:
I added this as a double check that the content store stored the data in the same location as the cache did.  Do you think this check excessive?

Reply:
this does remind me, that if we leave this double-check in, it needs an errnum.  I decided on EPROTO.

Reply:
Oh, right: we computed the hash over the blob at the same time as sending the store request so that we could pre-create the (empty) cache entry.  It does seem a bit excessive given that both ends should be using the same hash algorithm, but I'm fine if you prefer to leave it in - maybe just add a comment?

Reply:
I'll also leave a note that it could be excessive.  I recall a long time ago I accidentally appended a NUL byte to some data and it caused me hell debugging it.  So perhaps I'm being overly worrisome.